### DISCUSSION TITLE
API Naming Help

### DISCUSSION THREAD

**Discussion by njfio** [langflow-ai/langflow#1966] [Ideas]
Hi, 

I love the depth of the configuration options in Langflow and how the new toggle button for tweaks builds the entire json structure.    That said, there isn't a description field in the json that helps to identify what the component is doing. 

When you click on Tweaks in the API window, it's so easy to understand what component does what.
![CleanShot 2024-05-24 at 09 27 18](https://github.com/langflow-ai/langflow/assets/7220/3146e703-b9b1-4732-afd3-ad9d9fd4e1df)

But there is nothing to tell you what the components are from the configuration.

![CleanShot 2024-05-24 at 09 28 31](https://github.com/langflow-ai/langflow/assets/7220/f9164155-6151-4ba7-b900-6b79d7174bea)

![CleanShot 2024-05-24 at 09 29 26](https://github.com/langflow-ai/langflow/assets/7220/c35c25c6-d41d-49f6-9d0f-4a05dfe2ed1c)

This is a challenge for my application, which is designed to make it easy to invoke, chain, and consume Langflow workflows from the command line.   

Can a description/name field be added to all the tweaks override settings so that it's possible to understand what each component is intended to do just from the configuration file?   Ideally, the default description aligns with the accordion drop-downs in the API window (screenshot above).

Now, when I invoke my command and populate the autocomplete overrides, it pulls key names
![CleanShot 2024-05-24 at 09 34 57@2x](https://github.com/langflow-ai/langflow/assets/7220/58ac741c-8b0e-47eb-aab5-5a95a21cb5fc)

If something like the above was implemented, I could change it so the values in the autocomplete are pulled from the description
- instead of seeing - "TextInput-P3cGK.template" it would be "Workers Context.template". 

I can implement some level of a solution for this within my application, but it seems like something that might be useful in the Langflow project as a whole.  

Thanks,
Nick

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
